src/Workspaces/Core/Portable/Options/SerializableOptionSet.cs

```diff
@@ -76,14 +76,33 @@ internal sealed partial class SerializableOptionSet : OptionSet
         {
         }
 
+        /// <summary>
+        /// Returns an option set with all the serializable option values prefetched for given <paramref name="languages"/>,
+        /// while also retaining all the explicitly changed option values in this option set for any language.
+        /// </summary>
         public SerializableOptionSet WithLanguages(ImmutableHashSet<string> languages)
```

Comment:
>WithLanguages [](start = 37, length = 13)

Can we also move the call to this method out of Branch method, to the methods that calculate idToProjectStateMap?

`options ??= Options.WithLanguages(GetProjectLanguages(idToProjectStateMap));`

Perhaps we can also make it more efficient - it's currently enumerating all projects and creating a hash set.

Reply:
Sure, I can do that as a follow-up PR as it seems like just a performance optimization?

Reply:
ok

---
In reply to: [392523056](https://github.com/dotnet/roslyn/pull/42388#discussion_r392523056) [](ancestors = 392523056)

Reply:
I will do this as a follow-up change and merge this to unblock your work.